src/Common/ZooKeeper/ZooKeeperCommon.cpp

```diff
@@ -898,4 +898,31 @@ ZooKeeperRequestFactory::ZooKeeperRequestFactory()
     registerZooKeeperRequest<OpNum::FilteredList, ZooKeeperFilteredListRequest>(*this);
 }
 
+PathMatchResult matchPath(std::string_view path, std::string_view match_to)
+{
+    using enum PathMatchResult;
+
+    if (match_to == "/")
+        return path == "/" ? EXACT : IS_CHILD;
+
+    const auto clean_path = [](auto & p)
+    {
+        if (p.ends_with('/'))
+            p.remove_suffix(1);
+    };
+
+    clean_path(path);
+    clean_path(match_to);
```

Comment:
Also maybe much shorter without such complications:
```suggestion
   if (path.ends_with('/'))
        path.remove_suffix(1);
   if (match_to.ends_with('/'))
        match_to.remove_suffix(1);
```